main.tf

```diff
@@ -47,7 +47,7 @@ resource "google_compute_subnetwork" "subnetwork" {
   enable_flow_logs         = lookup(var.subnets[count.index], "subnet_flow_logs", "false")
   network                  = google_compute_network.network.name
   project                  = var.project_id
-  secondary_ip_range       = var.secondary_ranges[lookup(var.subnets[count.index], "subnet_name", null)]
+  secondary_ip_range       = [for i in range(length(contains(keys(var.secondary_ranges), var.subnets[count.index]["subnet_name"]) == true ? var.secondary_ranges[var.subnets[count.index]["subnet_name"]] : [])) : var.secondary_ranges[var.subnets[count.index]["subnet_name"]][i]]
```

Comment:
I'm not sure this is the right PR for this comment, but  I wanted to raise an idea.  Please feel free to let me know if I should raise this in an issue instead 😄    Rather then use a for loop here, what about the idea of using a dynamic block, and getting rid of the `secondary_ranges` data structure all together, and replacing it with a `secondary_ranges` list of maps in the `subnets` map?

Reply:
I think that's a good idea and would be happy to look at a PR.